File: Problems/22-Merge-Strings-Alternately/Merge-Strings-Alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.length();
        int m = word2.length();
        string s = "";
        
        int i = 0, j = 0;
        while (i < n and j < m)
        {
            s += word1[i++];
            s += word2[j++];
        }
        if (i == n)
        {
            while (j < m) s+= word2[j++];
        }
        if (j == m)
        {
            while (i < n) s+= word1[i++];
        }
        return s;
    }
};

File: Problems/22-Merge-Strings-Alternately/Merge-Strings-Alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        l1 = len(word1)
        l2 = len(word2)
        i,j = 0,0
        res = list()
        while i < l1 or j < l2:
            if i < l1:
                res.append(word1[i])
                i += 1
            if j < l2:
                res.append(word2[j])
                j += 1
        
        return "".join(res)